### DISCUSSION TITLE
Seeking more information on the Univariate Motif forecasting model

### DISCUSSION THREAD

**Discussion by yerbles** [winedarksea/AutoTS#145] [Q&A]
Hi,

Thanks for putting together this excellent package! In my work I do a lot of forecasting with energy data and I have found that the Univariate/Multivariate Motif model performs quite well just on its own. I am wanting to get a better understanding of how this algorithm functions and noticed that the "MotifSimulation" class is described as 

- "More dark magic created by the evil mastermind of this project. Basically a highly-customized KNN."

Are there any particular references that were used when writing this code that you could point me to?
I'd like to understand:
1. How the Motifs are selected;
2. How the multivariate version compares to the univariate;
3. How the Motifs are subsequently used in presumably KNN regression.

Thanks so much for your help!
LY

**Comment by winedarksea** [comment]
There were no references for this. My starting point was the idea that "history repeats itself" and I created a model to do that - MotifSimulation. After having created that, I realized it was basically KNN and simplified it down to a faster version. And then again recently for the Metric and Seasonality Motifs (next release). 

1. Motifs are selected by a distance metric. Basically how numerically similar they are. In the MetricMotif it is quite simply MAE. In seasonality motify the datepart of the index is used and then compared. The K closest neighbors are choosen (or more accurately, the part of those neighbors FOLLOWING the comparison window). Then the point_method is how they are averaged to the center forecast.
2.  Multivariate motif compares across all time series present - so perhaps another time series had a very similar pattern which can be used, even if the target series has never had that pattern before. Each series forecast is still unique, though. Sectional motif is similar but it takes the distance metric across all time series at once, thus a 'market' view and all series use the same time indexes for the K neighbors.
3.  Kinda already answered above, but there's no regression here (I've also used them as features, but here they are not) just taking the post period window. 
It is my belief that good preprocessing is critical for this model, which is one of the reasons it does well here. I have lots of preprocessing.
